handle/Timer.lua

```diff
@@ -1,37 +1,210 @@
 require('CeresStdLib.base.Native')
 require('CeresStdLib.base.Log')
-local free_timers = {}
+
+local Timer 		= {}
+local free_timers	= {}
+
+FLAG_TIMER_PAUSED	= 1
+FLAG_TIMER_ALTERED	= 2
+FLAG_TIMER_RESUMED	= 4
+
+free_timers.free	= 0
+free_timers.has = function() return free_timers.free > 0 end
+free_timers.push = function(t)
+	if not free_timers[GetHandleId(t)] then
+		free_timers.free 				= free_timers.free + 1
+		free_timers[GetHandleId(t)] 	= true
+		free_timers[free_timers.free]	= t
+		return true
+	end
+	return false
+end
+free_timers.pop = function()
+	local obj
+	if free_timers.free > 0 then
+		obj 							= free_timers[free_timers.free]
+		free_timers[GetHandleId(obj)] 	= nil 
+		free_timers[free_timers.free]	= nil
+		free_timers.free 				= free_timers.free - 1
+	end
+	return obj
+end
 
 replaceNative('CreateTimer', function()
-	k, v = next(free_timers)
-	if k ~= nil then
-		free_timers[k] = nil
-		return k
+	local obj
+	if free_timers.has() then
+		obj = free_timers.pop()
 	else
-		return Native.CreateTimer()
+		obj = Native.CreateTimer()
+	end
+	Timer[GetHandleId(obj)]      = {
+		hasCallback = false,
+		looped      = false,
+		running     = false,
+		inCallback  = false,
+		callback    = nil,
+		pauseFlag   = 0,
+		duration    = 0.,
+		elapsed     = 0.
+	}
+	return obj
+end)
+
+replaceNative('PauseTimer', function(t)
+	local i = GetHandleId(t)
+	if Timer[i].running then
+		Timer[i].running    = nil
+		Timer[i].pauseFlag  = (BlzBitOr(Timer[i].pauseFlag, FLAG_TIMER_PAUSED))
```

Comment:
It seems that WC run Lua 5.3, which [has bitwise operation implemented natively](https://www.lua.org/manual/5.3/manual.html#3.4.2). Check and replace `BlzBitOr()`.

Reply:
Didn't want to use them at first, so kept those bit operations. Replacing them later.